src/targets/github.ts

```diff
@@ -395,32 +413,37 @@ export class GithubTarget extends BaseTarget {
           Accept: DEFAULT_CONTENT_TYPE,
         },
       });
+    } catch (e) {
+      throw new Error(
+        `Cannot get asset on GitHub. Status: ${(e as any).status}\n` + e
+      );
+    }
 
+    const etag = response.headers['etag'];
+    if (etag && etag.length > 0) {
       // ETag header comes in quotes for some reason so strip those
-      const remoteChecksum = response.headers['etag']?.slice(1, -1);
-      if (remoteChecksum) {
-        // XXX: This local checksum calculation only holds for simple cases:
-        // https://github.com/getsentry/craft/issues/322#issuecomment-964303174
-        const localChecksum = createHash('md5').update(file).digest('hex');
-        if (localChecksum !== remoteChecksum) {
-          logger.trace(`Checksum mismatch for "${name}"`, response.headers);
-          throw new Error(
-            `Uploaded asset MD5 checksum does not match local asset checksum for "${name} (${localChecksum} != ${remoteChecksum})`
-          );
-        }
-        uploadSpinner.succeed(`Uploaded asset "${name}".`);
-      } else {
-        logger.warn(`Response headers:`, response.headers);
-        uploadSpinner.succeed(
-          `Uploaded asset "${name}, but failed to find a remote checksum. Cannot verify uploaded file is as expected.`
-        );
-      }
-      return url;
-    } catch (e) {
-      uploadSpinner.fail(`Cannot upload asset "${name}".`);
+      return etag.slice(1, -1);
+    }
 
-      throw e;
+    return await this.md5FromUrl(url);
+  }
+
+  private async md5FromUrl(url: string): Promise<string> {
+    this.logger.debug('Downloading asset from GitHub to check MD5 hash: ', url);
+    let response;
+    try {
+      response = await this.github.request(`GET ${url}`, {
+        headers: {
+          Accept: DEFAULT_CONTENT_TYPE,
+        },
+      });
+    } catch (e) {
+      throw new Error(
+        `Cannot download asset from GitHub. Status: ${(e as any).status}\n` + e
```

Comment:
How hard is it to put a retry around this? Intermittent network issues seem likely enough that I feel like we're asking for randomly failed releases if we aren't a little more graceful here.

Reply:
@iker-barriocanal and I talked about the ideal scenario of being able to gather metrics around this and other code paths -- to focus effort on what needs improvement.

We've also talked about retries.

For the cases we've seen recently, retry on this level would not have helped. E.g. in https://github.com/getsentry/publish/runs/4222412947?check_suite_focus=true#step:8:427, retrying on 404 would just take longer to fail the overall run. Retrying by rerunning the target fixed the problem though, because it re-uploads assets.

So seems like a more holistic architectural decision around retries would be what we need, something we didn't intend to own right now.

@chadwhitacre || @iker-barriocanal do you have in mind cases where a retry in this level would help? If you do have concrete cases, then by all means we can add retries.

Reply:
Nope. You've thought it through further than I have. Suggestion withdrawn. :)

Reply:
Late to the game but we already have a retry plugin on Octokit here which retries on 5xx errors automatically. Retrying on 404s doesn't make much sense.